src/context.rs

```diff
@@ -70,15 +73,15 @@ impl<'a> Context<'a> {
             .map(|(a, b)| (*a, b.vals.first().cloned().unwrap().into_string().unwrap()))
             .collect();
 
-        // TODO: Currently gets the physical directory. Get the logical directory.
-        let current_dir = Context::expand_tilde(dir.into());
-
+        let current_dir = Context::expand_tilde(path.into());
+        let logical_dir = logical_path;
         let shell = Context::get_shell();
 
         Context {
             config,
             properties,
             current_dir,
+            logical_dir,
             dir_contents: OnceCell::new(),
```

Comment:
**thought:**
It looks like we're starting to have a good number of directory-specific properties in `Context`. It may be sensible to create a `Directory` struct within `Context` to house them.

Reply:
As a matter of fact, current_dir & dir_contents are used by a variety of modules and are "general" purpose stuff (like repo is to the directory module). As I read it, only logical_dir is directory module specific.
Well, you call really...